ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -239,6 +240,9 @@ public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> stor
     checkStarted();
     // allows concurrent gets
     final Timer.Context context = metrics.getResponse.time();
+    if (ids.size() > 1 && ids.size() != new HashSet<>(ids).size()) {
```

Comment:
While it is good to have the check here in `get()`, I feel that this check really belongs in `put()` (and `delete()`). The store should guard against duplicates in `MessageWriteSet` as it takes it to an incorrect state. There is nothing inherently wrong with duplicate keys in `get()` (though it makes sense to nip it in the bud in a replication flow). What do you think?

Reply:
That is true. I will add a check in the PUT too. However, the GET check is required too because the PUT check is "too late". The PUT will reject the operation but the same operation will be endlessly repeated.

Reply:
Sure, I am in favor of keeping the check in `get()`. Is it required even with the change in `handleGetRequest()` to deduplicate?

Reply:
It's not required. However since it is a public API, I think it can perform its own checks and fail since there should be no reason to ask for the same blob twice in the same request